crates/bevy_transform/src/components/global_transform.rs

```diff
@@ -24,249 +24,43 @@ use std::ops::Mul;
-        self.local_x()
-    }
-
-    /// Get the unit vector in the local y direction
-    #[inline]
-    pub fn local_y(&self) -> Vec3 {
-        self.rotation * Vec3::Y
-    }
-
-    /// Equivalent to [`local_y()`][GlobalTransform::local_y]
-    #[inline]
-    pub fn up(&self) -> Vec3 {
-        self.local_y()
-    }
-
-    /// Equivalent to [`-local_y()`][GlobalTransform::local_y]
-    #[inline]
-    pub fn down(&self) -> Vec3 {
-        -self.local_y()
-    }
-
-    /// Get the unit vector in the local z direction
-    #[inline]
-    pub fn local_z(&self) -> Vec3 {
-        self.rotation * Vec3::Z
-    }
-
-    /// Equivalent to [`-local_z()`][GlobalTransform::local_z]
-    #[inline]
-    pub fn forward(&self) -> Vec3 {
-        -self.local_z()
-    }
-
-    /// Equivalent to [`local_z()`][GlobalTransform::local_z]
-    #[inline]
-    pub fn back(&self) -> Vec3 {
-        self.local_z()
-    }
-
-    #[doc(hidden)]
-    #[inline]
-    pub fn rotate(&mut self, rotation: Quat) {
-        self.rotation = rotation * self.rotation;
-    }
-
-    #[doc(hidden)]
-    #[inline]
-    pub fn rotate_around(&mut self, point: Vec3, rotation: Quat) {
-        self.translation = point + rotation * (self.translation - point);
-        self.rotation *= rotation;
-    }
-
-    /// Multiplies `self` with `transform` component by component, returning the
-    /// resulting [`GlobalTransform`]
-    #[inline]
-    #[must_use]
-    pub fn mul_transform(&self, transform: Transform) -> Self {
-        let translation = self.mul_vec3(transform.translation);
-        let rotation = self.rotation * transform.rotation;
-        let scale = self.scale * transform.scale;
-        Self {
-            translation,
-            rotation,
-            scale,
-        }
-    }
-
-    /// Returns a [`Vec3`] of this [`Transform`] applied to `value`.
-    #[inline]
-    pub fn mul_vec3(&self, mut value: Vec3) -> Vec3 {
-        value = self.scale * value;
-        value = self.rotation * value;
-        value += self.translation;
-        value
+    /// Retrieves the [`Transform`] inner type of `Self`
+    pub fn into_inner(self) -> Transform {
+        self.0
     }
+}
 
-    #[doc(hidden)]
-    #[inline]
-    pub fn apply_non_uniform_scale(&mut self, scale: Vec3) {
-        self.scale *= scale;
-    }
+impl Deref for GlobalTransform {
+    type Target = Transform;
 
-    #[doc(hidden)]
-    #[inline]
-    pub fn look_at(&mut self, target: Vec3, up: Vec3) {
-        let forward = Vec3::normalize(self.translation - target);
-        let right = up.cross(forward).normalize();
-        let up = forward.cross(right);
-        self.rotation = Quat::from_mat3(&Mat3::from_cols(right, up, forward));
+    fn deref(&self) -> &Self::Target {
+        &self.0
     }
 }
 
-impl Default for GlobalTransform {
-    fn default() -> Self {
-        Self::identity()
+impl DerefMut for GlobalTransform {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
     }
 }
```

Comment:
Trait implementations are public, so even though the inner field is only `pub(crate)`, this implementation allows it to be modified by any crate.